source/extensions/quic_listeners/quiche/platform/quic_mem_slice_span_impl.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "envoy/buffer/buffer.h"
+
+#include "common/common/stack_array.h"
+
+#include "quiche/quic/core/quic_types.h"
+#include "quiche/quic/platform/api/quic_mem_slice.h"
+#include "quiche/quic/platform/api/quic_string_piece.h"
+
+namespace quic {
+
+// Wrap a Buffer::Instance and deliver its data to quic stream through as less
+// copy as possible.
```

Comment:
Wraps a Buffer::Instance, delivering its data with the minimum number of copies.